### DISCUSSION TITLE
How to use enums?

### DISCUSSION THREAD

**Discussion by Zer0xxxx** [alfateam/rdb#96] [Q&A] [answered]
Hello!
First of all, thank you a lot for this fp ORM.

I would like to know how to use enums with Orange ORM?

From the docs, we have:
string, numeric, boolean, uuid, date...

But I don't see any clue about enums.
Should I do a trick with a validation function?

I would like to have a `locales` enum like so:
```sql
CREATE TYPE LOCALES_ENUM AS ENUM ('en-US', 'fr', 'it');
```

Should I do this as a workaround?

```ts
function validateLocale(value: MaybeUndefined<MaybeNull<string>>) {
  if (value === undefined || value === null) return;
  if (!(locales as any).includes(value)) throw new Error('Unknown locale');
}

const map = orange
  .map((x) => ({
    team: x.table('team').map(({ column }) => ({
      id: column('id').numeric().primary().notNullExceptInsert(),
      lang: column('lang').string().default(baseLocale).validate(validateLocale)

// * ...
```

**Comment by lroal** [comment, marked as answer]
Enums are not supported today.
Yes, it makes sense to support it.

A validate function will solve it.

**Reply by Zer0xxxx** [reply]
Thank you a lot!
